### ISSUE TITLE
uploading files

### ISSUE THREAD

**Issue by ArneNostitz** [TheFehr/n8n-nodes-pocketbase#6]
i am trying to get a file uploaded and was wondering how to do this with the pb node

thanks!

**Comment by TheFehr** [comment]
I have not had that need, it might not be supported as of now.

**Comment by ArneNostitz** [comment]
would be great if it did :) 

it would need a more complex property management though (binary files, field-name) - send as form-data. 

otherwise your plugin is very convenient :)

**Comment by ArneNostitz** [comment]
sadly i am really hitting a wall with PB and data upload ... maybe its not PB. but i am not able to get it to work. 

are you using the JS SDK in the background? or are you building http requests?

**Comment by TheFehr** [comment]
The node uses the sdk in the background

**Comment by javierfuentesm** [comment]
@ArneNostitz, did you find a workaround? I'm facing the same issue

**Comment by ArneNostitz** [comment]
yes. using the http request node in form data mode, sending data as binary - that works. its a bit tricky because i used to test by stepping through the workflow and pinning the outputs - but that does not work for binary - it looks like it does but is actually not sending data - and that creates errors. 
the workflow needs to run, then it works

**Comment by javierfuentesm** [comment]
> yes. using the http request node in form data mode, sending data as binary - that works. its a bit tricky because i used to test by stepping through the workflow and pinning the outputs - but that does not work for binary - it looks like it does but is actually not sending data - and that creates errors. the workflow needs to run, then it works

Thank you and how did you authenticate your request ?

**Comment by ArneNostitz** [comment]
depends on what you want to do. if you just have one user, you can use the basic auth with username, password set in n8n. 

otherwise you need to auth through another http request (login) and then use the token

<img width="557" alt="image" src="https://github.com/user-attachments/assets/6bf3c81c-16fe-4157-8baa-1fd5cf94ac67" />
<img width="630" alt="image" src="https://github.com/user-attachments/assets/85e5ff85-a71e-41fd-8838-80deca3306cd" />

**Comment by javierfuentesm** [comment]
> depends on what you want to do. if you just have one user, you can use the basic auth with username, password set in n8n.
> 
> otherwise you need to auth through another http request (login) and then use the token
> 
> <img alt="image" width="557" src="https://private-user-images.githubusercontent.com/16592596/401694654-6bf3c81c-16fe-4157-8baa-1fd5cf94ac67.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0NTc4MzMsIm5iZiI6MTczNjQ1NzUzMywicGF0aCI6Ii8xNjU5MjU5Ni80MDE2OTQ2NTQtNmJmM2M4MWMtMTZmZS00MTU3LThiYWEtMWZkNWNmOTRhYzY3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDIxMTg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjMmMzZDM0ODRjZGU1ODA4YzE2NWU0YWM5ZGQ4YWQzMDUxOGExNTZhYmY3Mzg3Zjc1NTM1YjAxMmU0N2FkMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.E8h11H-cVYZDEPYFK-2bk6u2F42ow9Ge-Syhjn23M5c"> <img alt="image" width="630" src="https://private-user-images.githubusercontent.com/16592596/401694958-85e5ff85-a71e-41fd-8838-80deca3306cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY0NTc4MzMsIm5iZiI6MTczNjQ1NzUzMywicGF0aCI6Ii8xNjU5MjU5Ni80MDE2OTQ5NTgtODVlNWZmODUtYTcxZS00MWZkLTg4MzgtODBkZWNhMzMwNmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA5VDIxMTg1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExZmVlODMxNzAzMjg4NTU0Yzg3M2NlYmUxMjZhZDRjNDlhMGZkZDlmM2NhMGM5ZWYyZDU4ODgxMGNiZmY2ZjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yoTKg56iYl9ZTd3ZAbfvuV4Hjhxtp1IzC27_9AjMcz4">

you are a life saver thank you !

**Comment by TheFehr** [comment]
@javierfuentesm I just added a new `PocketBase - Send` node, which let's you send arbitrary authenticated request against your pocketbase it just uses the js sdk's pb.send method.
I assume that should let upload the files without doing the auth yourself.
